src/main.jsx

```diff
@@ -17,8 +18,22 @@ root.render(
       <Routes>
         <Route path="/" element={<App />}>
           <Route index element={<LandingPage />} />
-          <Route path="login" element={<LoginPage />} />
-          <Route path="register" element={<RegisterPage />} />
+          <Route
+            path="login"
+            element={
+              <AuthRedirectRoute>
+                <LoginPage />
+              </AuthRedirectRoute>
+            }
+          />
+          <Route
+            path="register"
+            element={
+              <AuthRedirectRoute>
+                <RegisterPage />
+              </AuthRedirectRoute>
+            }
+          />
           <Route
             path="analysis"
             element={<ProtectedRoute component={AnalysisPage} />}
```

Comment:
RegisterPage and LoginPage are wrapped in the AuthRedirectRoute component, AnalysisPage is passed to ProtectedRoute.

ProtectedRoute is waiting for confirmLogin to finish, I read that this method is preferred over wrapping the component, like as in where AuthRedirectRoute is only making a simple check for the logged in status

Reply:
Can you explain your second sentence here again? I'm not fully following

Reply:
Sorry @rknizzle 

I originally thought that only ProtectedRoutes was depending on the asynchronous confirmLogin function. And that AuthRedirectRoute was simply checking the state of isLoggedIn.

It was my understanding that if a component relied on an async function to render, then passing that component to ProtectedRoutes is the best option. 

If a component wasn't relying on an async function to render, you can just wrap it around the component and render it like I did with login and register.

But now that I am going over this again. 

Don't all of these technically rely on confirmLogin()? 

...

With the analysis page, I just know for a fact that it has to wait for confirmLogin to finish, because before I added the finally block to the confirmLogin function to have protectedroute wait for it to finish, I was getting all kinds of issues with timing and renders.

Reply:
True yea I think they're kind of the same in that they both wait for confirmLogins result. They can probably both be written with the same pattern too whereas right now one of them takes in children and the other one takes in a component as a prop or something